tn_api_v2_core.yml

```diff
@@ -0,0 +1,956 @@
+                    description: The IATA code of the destination airport or city.
+                    type: string
+                    example: "YVR"
+                to_type:
+                    description: The type of the destination, "C" for City or "A" for Airport.
+                    type: string
+                    enum: [ "C", "A" ]
+                    example: "C"
+                departure_date:
+                    description: The departure date of the segment, format = YYYY-MM-DD.
+                    type: string
+                    example: "2023-06-30"
+                cabin_class:
+                    description: The cabin class of the segment.
+                    type: string
+                    enum: [ "E", "PE", "BC", "FC", "PFC" ]
+                    example: "E"
+
+
+        MinimalGenerateSolutionsRequest:
+            type: object
+            title: Generate Solutions Request (Minimal)
+            properties:
+                trip_id:
+                    description: The trip id of the search. This will be used in all future requests relating to this search.
+                    type: string
+                    example: "fb3d8eff432a329dc21fad8449c686fa93e5ce24"
+                datasource_responses:
+                    description: The list of datasource responses to be used to generate solutions.
+                    type: object
+                    allOf:
+                        -   $ref: '#/components/schemas/MinimalDatasourceResponse'
+
+
+        FullGenerateSolutionsRequest:
+            type: object
+            title: Generate Solutions Request (With Extra Details)
+            properties:
+                trip_id:
+                    description: The trip id of the search. This will be used in all future requests relating to this search.
+                    type: string
+                    example: "fb3d8eff432a329dc21fad8449c686fa93e5ce24"
+                datasource_responses:
+                    description: The list of datasource responses to be used to generate solutions.
+                    type: object
+                    allOf:
+                        -   $ref: '#/components/schemas/FullDatasourceResponse'
+
+        MinimalDatasourceResponse:
+            type: object
+            properties:
+                4f499298b303c55a1ee522118afce28abe2f68e4:
+                    description: The ID of the datasource request. The ID varies between requests.
+                    type: array
+                    items:
+                        $ref: "#/components/schemas/BasicGenerateDatasourceRequest"
+
+        FullDatasourceResponse:
+            type: object
+            properties:
+                4f499298b303c55a1ee522118afce28abe2f68e4:
+                    description: The ID of the datasource request. The ID varies between requests.
+                    type: array
+                    items:
+                        $ref: "#/components/schemas/FullGenerateDatasourceRequest"
```

Comment:
I think it would be good to link back to the first request here. 

The datasource_request_id of a datasource request returned from the first **get-searches** call after the OTA has performed the flight searches requested. The value of the id will be the results from that search. 

wdyt?

Reply:
way better!